CONTRIBUTING.md

```diff
@@ -119,6 +119,37 @@ Use the format:
 This helps provide context for future readers and keeps the TODO relevant and
 actionable as the project evolves.
 
+## Opening a pull request
```

Comment:
I think I'd remove this heading, and move everything from line 124 to just before the "Commit messages" bit. (I think it's important to start with "All code changes must go through a pull request.")

Beyond that, all the text looks good.

(I'm about to go offline, so if you want to get this in today, please ask another reviewer. But feel free to point them at this comment...)